common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -59,12 +59,14 @@ internal class ExceptionHandlingHelper
                         message: await GetExceptionMessageAsync(response).ConfigureAwait(false))
                 },
 
+#pragma warning disable CS0618 // This exception does not appear to be returned by the service for ServiceClient operations
```

Comment:
Is this being marked as obsolete because service code doesn't generate `HttpStatusCode.PreconditionFailed`? How sure are we of that? 
Does our SDK translate some error code to `HttpStatusCode.PreconditionFailed`?

Reply:
Actually this was a result of marking the DeviceMessageLockLostException as obsolete. But this is converting a specific Http Status to this exception so I'll be reverting the change and just documenting this in the exception.